### ISSUE TITLE
pay an extra product by adding it to the customer from the bo in an already paid order

### ISSUE THREAD

**Issue by alfonsolr09** [PrestaShop/PrestaShop#18803]
#### Describe the bug

Currently if you add a product to an order that has already been paid, the customer cannot pay the amount that remains to be paid

#### Expected behavior

When I add a product to an order that has already been paid, there should be an option to send an email to the customer so that they can pay the amount that remains to be paid.
#### Steps to Reproduce

1. Go to orders 
2. Click on in a correctly placed and paid order 
3. Add a product to an order already placed

**Screenshots**

https://youtu.be/mmlqsiVUpDI

#### Additional information

* PrestaShop version: 1.7.6.4
* PHP version: 7.2.29

**Comment by PierreRambaud** [comment]
Hi @alfonsolr09 and thanks for your report.

Not sure it's a bug or an improvement :thinking: 
Ping @PrestaShop/prestashop-product-team what do you think?

**Comment by alfonsolr09** [comment]
I know, I did not know how to classify it

**Comment by marionf** [comment]
It seems to be an improvement, as it's about order page I ping @MatShir

**Comment by MatShir** [comment]
@alfonsolr09, Have you try to create a manual order for your client and share the link of the checkout?

**Comment by alfonsolr09** [comment]
Yes, but I have to buy a product. I would like to generate a link where I select how much the client should pay me and generate the link and send it to them

something like this 
https://prnt.sc/s85chu

<img src="https://image.prntscr.com/image/MqfsYptbSAmytFiUmwwkBQ.png" />

**Comment by alfonsolr09** [comment]
<img src="https://image.prntscr.com/image/MqfsYptbSAmytFiUmwwkBQ.png" />

**Comment by alfonsolr09** [comment]
@marionf How is the topic going?

**Comment by MatShir** [comment]
Hi @alfonsolr09, 
Could you share us more about the context of the needs? Have you or your customer forgotten a product in the order? How do you proceed exactly and can you describe the steps you occur with your client?

The more information you share, the easier it will be for us to understand the need and prioritize it :smile: 

Just be carefully, legally, your client should not have 2 different versions of the invoices from the same order, we allow to edit an order but it should be done before the client paid and receive the invoice. If something has been forgotten, we highly recommend to send a credit slip and make a new invoice with the right content.

**Comment by alfonsolr09** [comment]
- I explain a little. 

A customer buys an item, processes the order and pays for it. 
Call me that same afternoon to add another item to your order, I add it, but you cannot pay for that item from that same order without generating a new one.


 - I tell you another example, with the same functionality but applied in another way.
 
An older client who does not know how to pay online, tells me that he wants me to send him a box of screws, but how does he pay me?  I propose that a function be created to be able to generate a link without generating an order to make the payment, in a box you select the amount of the purchase and a link is generated that I send to the client so that he only has to pay without having to  do nothing else. 


Ps: the invoice topic, adding an item to an already paid order would generate another invoice, (it really is as if you were making another order but everything from the same)

**Comment by MatShir** [comment]
Thanks for your explanation @alfonsolr09,

> An older client who does not know how to pay online, tells me that he wants me to send him a box of screws, but how does he pay me? I propose that a function be created to be able to generate a link without generating an order to make the payment, in a box you select the amount of the purchase and a link is generated that I send to the client so that he only has to pay without having to do nothing else.

For this part, I recommend you to create a manual order and share the link or send it by email that is located at the bottom of the page as you can see on the screenshot below. 
![Capture](https://user-images.githubusercontent.com/43613217/83126749-923c6380-a0d9-11ea-9cbc-29764c01c4b1.PNG)

> A customer buys an item, processes the order and pays for it.
> Call me that same afternoon to add another item to your order, I add it, but you cannot pay for that item from that same order without generating a new one.
                                                 
Here, I see the missing feature to finish your order. It looks durable since we provide a link to the checkout on the creation of a manual order. Your insight is shared to the product team but because there is a workaround, it might not be must-have for the next version. 
The project is open-source, so in the meantime, someone can provide a contribution but since there is a workaround. Or if you have the ability to contribute, we will be happy to review your work.

For the other user in the meantime, you could also by creating a manual order that will generate a link to the checkout if you have to make the order or add a product for your customer. 

The link on your screenshot where does it redirects?
I'm not confident with the free amount input, the amount should match the selected invoice. If discounts need to be added, it should first by apply to the order then updated on the invoice.

**Comment by alfonsolr09** [comment]
<img src="https://image.prntscr.com/image/gtGSvly4S1ex_mIQo4wRVA.png" />
the link redirects here where the customer can pay, by clicking send I suppose it redirects to a process similar to paying a cart in prestashop

**Comment by alfonsolr09** [comment]
I think it is a good way for a customer to pay you for an order that he places by phone or many other times.

**Comment by alfonsolr09** [comment]
@MatShir

**Comment by MatShir** [comment]
Your feature is validated with the team. But it won't be prioritized for the next version but however, PrestaShop is an open-source project, if you have the technical knowledge or find someone that could don't hesitate to propose a pull request ([find more information here](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/)). We will be happy to review your contribution to the PrestaShop core.
